test_environments/message_and_event_brokers/kafka/sasl_plain/with_tls/README.md

```diff
@@ -0,0 +1,72 @@
+# Overview
+This is a simple setup of Kafka with authentication and TLS encryption. 
+It is intended to be used to set up test environment in which Kafka brokers require clients to authenticate.
+All communication is done over secure TLS connection.
+
+These test environments provide set up for Kafka instance which requires clients to authenticate 
+over secure TLS connection.
+
+Test environments are location in directories:
+* `ssl_all_connections` - Broker certificate has SAN set to `0.0.0.0`
+* `ssl_localhost` - Broker certificate has SAN set to `localhost`
+
+All certificates required for TLS are already generated. **They are not intended to be used in production.**
+<br>Certificates in this environment, are used to verify the identity of the Kafka broker to the clients.
+Pre-generated certificates are valid through `30/3/2024`. 
+If you need to generate new certificates checkout the README 
+files in directories `ssl_all_connections` and `ssl_localhost`.
+
+
+## How to connect to Kafka broker
+To connect to the broker from the client use:
+* URL localhost:9092
+* one of the users
+
+    | User   | Password     |
+    |--------|--------------|
+    | admin  | admin-secret |
+    | alice  | alice-secret |
+
+Users which are available for clients are defined in the `kafka_jaas.conf` file.
+
+Add CA certificate to the client certificate pool before starting it.
+<br>This is required to verify the identity of the Kafka broker.
+<br>If you want to skip verification of the Kafka broker's identity, 
+you can instruct your client to skip verification of the certificate.
```

Comment:
This really depends on the method used.
One method is to add certificate to the Operating System's CA certificate pool which can be quite dangerous because that certificate pool is used by all applications on the machine. If private key associated with this CA certificate is stolen a malicious entity can use it to sign certificates of malicious web sites and harm the user of the machine (cheating browser to trust malicious sites etc.).
The other method is to add it to the in-memory certificate pool of the specific application when the app starts.
This is done through the code of the app.
Different libraries achieve this in different ways so devs should consult library's manual.
Some libraries even do not offer this feature, which as you can see is pretty bad as it pushes dev to add test certificate to machine's official certificate pool.

Reply:
I have update the docs to point this out.